internal/storage/chunk_uploader.go

```diff
@@ -0,0 +1,214 @@
+package storage
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"github.com/googlecloudplatform/gcsfuse/internal/logger"
+	"github.com/googlecloudplatform/gcsfuse/internal/storage/gcs"
+	"github.com/googlecloudplatform/gcsfuse/internal/storage/storageutil"
+	"google.golang.org/api/googleapi"
+)
+
+type UploaderState int
+
+const (
+	// Initialized is the state of successful initialization, no writes so far.
+	Initialized UploaderState = iota
+	// Uploading is the state when an asynchronous upload is in progress.
+	Uploading
+	// UploadError is the state when an Upload failed.
+	UploadError
+	// Closed is the state of an uploader which has been finalized.
+	Closed
+	// StorageCopyTimeoutDuration is how long the chunkUploader waits for the
+	// copy call to the uploader to return. If it takes longer, the
+	// upload call times out.
+	StorageCopyTimeoutDuration time.Duration = time.Second
```

Comment:
As discussed offline, let's remove the timeout logic from asyncChunkUpload and this const

Reply:
ok

Reply:
Done.